sql/src/main/java/io/crate/operation/tablefunctions/GenerateSeriesFunction.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.tablefunctions;
+
+import io.crate.action.sql.SessionContext;
+import io.crate.analyze.WhereClause;
+import io.crate.data.Bucket;
+import io.crate.data.Input;
+import io.crate.data.Row;
+import io.crate.data.RowN;
+import io.crate.metadata.*;
+import io.crate.metadata.table.StaticTableInfo;
+import io.crate.metadata.table.TableInfo;
+import io.crate.metadata.tablefunctions.TableFunctionImplementation;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.elasticsearch.cluster.service.ClusterService;
+
+import javax.annotation.Nullable;
+import java.util.*;
+
+public class GenerateSeriesFunction {
+
+    private static final String NAME = "generate_series";
+    private static final TableIdent TABLE_IDENT = new TableIdent(null, NAME);
+
+    public static void register(TableFunctionModule module){
+        module.register(NAME, new BaseFunctionResolver(
+            Signature.numArgs(2,3).and(Signature.withLenientVarArgs(Signature.ArgMatcher.NUMERIC))) {
+
+            @Override
+            public FunctionImplementation getForTypes(List<DataType> dataTypes) throws IllegalArgumentException {
+                FunctionIdent fi = new FunctionIdent(NAME, dataTypes);
+
```

Comment:
to simplify this, it would probably make sense to cast the inputs to a common type in normalizeSymbol and have 4 static instances for int, long, float, double

also datatype specific tests are missing afaiks